controllers/actions.github.com/autoscalingrunnerset_controller_test.go

```diff
@@ -197,6 +177,46 @@ var _ = Describe("Test AutoScalingRunnerSet controller", Ordered, func() {
 			Expect(err).NotTo(HaveOccurred(), "failed to list EphemeralRunnerSet")
 			Expect(len(runnerSetList.Items)).To(BeEquivalentTo(1), "Only one EphemeralRunnerSet should be created")
 		})
+
+		It("It should read GitHub config secret from controller namespace if ReadGitHubConfigSecretFromControllerNamespace is enabled", func() {
+			err := k8sClient.Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: configSecret.Name, Namespace: autoscalingNS.Name}})
+			Expect(err).ToNot(HaveOccurred(), "failed to delete github-config-secret")
+
+			controller2 := controller
+			controller2.ReadGitHubConfigSecretFromControllerNamespace = true
+			controller2.ControllerNamespace = controllerNS.Name
+
+			err = controller2.SetupWithManager(mgr)
+			Expect(err).NotTo(HaveOccurred(), "failed to setup controller")
+
+			controllerNamespaceConfigSecret := createDefaultSecret(GinkgoT(), k8sClient, controller2.ControllerNamespace)
+			runner := createAutoscalingRunnerSet("secret-test-asrs", autoscalingNS, buildVersion, controllerNamespaceConfigSecret.Name, 0, 1)
+			err = k8sClient.Create(ctx, runner)
+			Expect(err).NotTo(HaveOccurred(), "failed to create AutoscalingRunnerSet")
+
+			created := new(v1alpha1.AutoscalingRunnerSet)
+			// Check if runner scale set is created on service
+			Eventually(
+				func() (string, error) {
+					err := k8sClient.Get(ctx, client.ObjectKey{Name: runner.Name, Namespace: autoscalingRunnerSet.Namespace}, created)
+					if err != nil {
+						return "", err
+					}
+
+					if _, ok := created.Annotations[runnerScaleSetIdAnnotationKey]; !ok {
+						return "", nil
+					}
+
+					if _, ok := created.Annotations[AnnotationKeyGitHubRunnerGroupName]; !ok {
+						return "", nil
+					}
+
+					return fmt.Sprintf("%s_%s", created.Annotations[runnerScaleSetIdAnnotationKey], created.Annotations[AnnotationKeyGitHubRunnerGroupName]), nil
+				},
+				autoscalingRunnerSetTestTimeout,
+				autoscalingRunnerSetTestInterval).Should(BeEquivalentTo("1_testgroup"), "RunnerScaleSet should be created/fetched and update the AutoScalingRunnerSet's annotation")
+
+		})
```

Comment:
i've only added this test on the autoscalingrunnerset controller so far because I wanted to get some feedback before writing similar tests on the other controllers. I struggled to find a better way to write this test, it doesn't seem ideal that I'm setting up a second controller in this test but one is already setup in the `BeforeEach` but it isn't configurable. Any thoughts on this would be appreciated.